micro_sam/evaluation/benchmark_datasets.py

```diff
@@ -0,0 +1,448 @@
+        ),
+        "sponge_em": lambda: datasets.sponge_em.get_sponge_em_data(
+            path=os.path.join(path, "sponge_em"), download=True,
+        ),
+        "vnc": lambda: datasets.vnc.get_vnc_data(
+            path=os.path.join(path, "vnc"), download=True,
+        ),
+        "asem_mito": lambda: datasets.asem.get_asem_data(
+            path=os.path.join(path, "asem"), volume_ids=datasets.asem.ORGANELLES["mito"], download=True,
+        )
+    }
+
+    if dataset_choice is None:
+        dataset_choice = available_datasets.keys()
+    else:
+        if not isinstance(dataset_choice, list):
+            dataset_choice = [dataset_choice]
+
+    for choice in dataset_choice:
+        if choice in available_datasets:
+            available_datasets[choice]()
+        else:
+            raise ValueError(f"'{choice}' is not a supported choice of dataset.")
+
+    return dataset_choice
+
+
+def _extract_slices_from_dataset(path, dataset_choice):
+    if dataset_choice in LM_2D_DATASETS or EM_2D_DATASETS:
+        ndim, tile_shape = 2, (512, 512)
+    else:
+        ndim, tile_shape = 3, (32, 512, 512)
+
+    available_datasets = {
+        # Light Microscopy datasets
+        "livecell": lambda: datasets.livecell._get_livecell_paths(path=path, split="test"),
+        "deepbacs": lambda: datasets.deepbacs._get_deepbacs_paths(path=path, split="test", bac_type="mixed"),
+    }
+
+    image_paths, gt_paths = available_datasets[dataset_choice]()
+    if dataset_choice in DATASET_RETURNS_FOLDER:
+        image_paths = glob(os.path.join(image_paths, DATASET_RETURNS_FOLDER[dataset_choice]))
+        gt_paths = glob(os.path.join(gt_paths, DATASET_RETURNS_FOLDER[dataset_choice]))
+
+    image_paths, gt_paths = natsorted(image_paths), natsorted(gt_paths)
+    assert len(image_paths) == len(gt_paths)
+
+    # Directory where we store the extracted ROIs.
+    save_image_dir = os.path.join(path, f"roi_{ndim}d", "inputs")
+    save_gt_dir = os.path.join(path, f"roi_{ndim}d", "labels")
+    if os.path.exists(save_image_dir) and os.path.exists(save_gt_dir):
+        return save_image_dir, save_gt_dir, ndim
+
+    os.makedirs(save_image_dir, exist_ok=True)
+    os.makedirs(save_gt_dir, exist_ok=True)
+
+    # Logic to extract relevant patches for inference
+    image_counter = 1
+    for image_path, gt_path in tqdm(
+        zip(image_paths, gt_paths), total=len(image_paths),
+        desc=f"Extracting patches for {dataset_choice}"
+    ):
+        image = imageio.imread(image_path)
+        gt = imageio.imread(gt_path)
+
+        if len(np.unique(gt)) == 1:  # There could be labels which does not have any annotated foreground.
+            continue
+
+        tiling = blocking([0, 0], gt.shape, tile_shape)
+        n_tiles = tiling.numberOfBlocks
+        tiles = [tiling.getBlock(tile_id) for tile_id in range(n_tiles)]
+        crop_boxes = [[tile.begin[1], tile.begin[0], tile.end[1], tile.end[0]] for tile in tiles]
```

Comment:
Again, needs some update for 3d.

Reply:
It would be easiest if you just create a list of `tuple[slice]` here directly and then use it wherever relevant below. This would enable 3d easily and also reduce confusion with flipping around the axis orders below.